src/test/java/subway/StationAcceptanceTest.java

```diff
@@ -59,71 +57,89 @@ void selectStationsTest() {
         ExtractableResponse<Response> response = 지하철역_조회();
 
         // then
-        assertThat(response.statusCode()).isEqualTo(HttpStatus.OK.value());
+        요청_결과_상태_검증(response, HttpStatus.OK.value());
 
         // then
+        지하철_역_개수_검증(response, 2);
+    }
+
+    private static void 지하철_역_개수_검증(ExtractableResponse<Response> response, int targetCount) {
         List<String> stationNames = response.jsonPath().getList("name", String.class);
 
-        assertThat(stationNames.size()).isEqualTo(2);
+        assertThat(stationNames.size()).isEqualTo(targetCount);
     }
 
     /**
      * Given 지하철역을 생성하고
      * When 그 지하철역을 삭제하면
      * Then 그 지하철역 목록 조회 시 생성한 역을 찾을 수 없다
      */
-    // TODO: 지하철역 제거 인수 테스트 메서드 생성
     @DisplayName("지하철역을 제거한다.")
     @Test
-    void deleteStationTest() {
+    void deleteStation() {
         // given
         ExtractableResponse<Response> gangnamStationCreationRes = 지하철역_생성(강남역);
-        ExtractableResponse<Response> yeoksamStationCreationRes = 지하철역_생성(역삼역);
+        ExtractableResponse<Response> YeosamStationCreationRes = 지하철역_생성(역삼역);
 
-        String stationId = gangnamStationCreationRes.jsonPath().getString("id");
+        String stationUrl = gangnamStationCreationRes.response().getHeader("Location");
```

Comment:
응답을 바로 받는 방법은 없을까요? gangnamStationCreationRes 를 꼭 거쳐야 하나요? 💭

Reply:
@pch8388 응답을 바로 받는 방법이라는 것이 잘 이해가 안됩니다! 아래 방식을 생각하신건가요?

```suggestion
        String stationUrl = 지하철역_생성(강남역).response().getHeader("Location");
```

gangnamStationCreationRes를 명시적으로 만드는 이유는 아래와 같이 인수 테스트의 then(검증) 코드를 두개의 ExtractableResponse<Response>로 표현하기 위해서입니다. E2E 인수 테스트 특성상 when과 then을 위한 api가 한개씩 존재해야한다고 생각했어요. 이에 검증 메소드도 두 api의 응답을 매개변수로 받아야한다고 생각했습니다.  


```
private static void 역_삭제_여부_검증(ExtractableResponse<Response> creationRes, ExtractableResponse<Response> getStationRes) {
        String 삭제된_역 = creationRes.jsonPath().getString("name");

        assertThat(getStationRes.jsonPath().getList("name")).doesNotContain(삭제된_역);
    }
```